includes/results.inc

```diff
@@ -1021,7 +1021,11 @@ class IslandoraSolrFacets {
       $link['count'] = $count;
       $link['link_plus'] = '<a' . drupal_attributes($attributes['plus']['attr']) . '>+</a>';
       $link['link_minus'] = '<a' . drupal_attributes($attributes['minus']['attr']) . '>-</a>';
-      $buckets[] = $link;
+      $buckets[$bucket] = $link;
+    }
+
+    if (isset($this->settings['solr_field_settings']['sort_by']) && $this->settings['solr_field_settings']['sort_by'] == 'index') {
+      ksort($buckets);
```

Comment:
Running this way... this would return the top number of results by count, ordered by value; while, it may be expected that it gives the top values (how I first read it, anyway). For example, say you're limiting the number of values returned, and by numbers, would only get things near the bottom of things lexicographically... Could end up with strange results.

May have to go jam things into the [`$params_array`](https://github.com/Islandora/islandora_solr_search/blob/7.x/includes/query_processor.inc#L188-L193)... Untested, but thinking something like:
```php
$facet_array = islandora_solr_get_fields('facet_fields', TRUE, FALSE);
[...]
$params_array = array(
  'facet' => 'true',
  'facet.mincount' => variable_get('islandora_solr_facet_min_limit', '2'),
  'facet.limit' => variable_get('islandora_solr_facet_max_limit', '20'),
  'facet.field' => explode(',', $facet_fields),
);
foreach ($facet_array as $field => $info) {
  $settings = $info['solr_field_settings'];
  if (($settings['sort_by'] == 'count' && $params_array['facet.limit'] < 0) ||
    ($settings['sort_by'] == 'index' && $params_array['facet.limit'] > 0)) {
    // Avoid specifying default behaviour:
    // @see https://wiki.apache.org/solr/SimpleFacetParameters#facet.sort
    // "The default is count if facet.limit is greater than 0, index otherwise."
    $params_array["facet.{$field}.sort"] = $info['sort_by'];
  }
}
```